### PR TITLE
DOC: Correct uniqueness of index for Series

### LINKED ISSUE #7808: Conflicting documentation about index uniqueness
The page http://pandas.pydata.org/pandas-docs/dev/generated/pandas.Series.html says:

> index : array-like or Index (1d)
> Values must be unique and hashable, same length as data. 

A little bit earlier the same page says: 

> Labels need not be unique but must be any hashable type.  

Other pages also mention non-unique index support (e.g. http://pandas.pydata.org/pandas-docs/dev/dsintro.html) It looks like `index` description should read:

> index : array-like or Index (1d)
> Values must be hashable, same length as data. Non-unique index values are allowed.

### PR DESCRIPTION
**Pull request by themrmax**
closes #7808

Just wanted to fix the docstring to reflect the fact that the index labels neither need to be unique ~~nor hashable~~.

### CODE REVIEW

**Comment by chris-b1** [comment]
It is true that labels no longer need to be unique, but they still must be hashable?

**Comment by themrmax** [comment]
No they don't need to be hashable currently:

```
In [5]: s = pd.Series ([1,2,3], index = ([1],[1],[1]))

In [6]: s
Out[6]:
[1]    1
[1]    2
[1]    3
dtype: int64
```

Do you think we should add a check for hashable?

**Comment by jreback** [comment]
try to index that and see what happens

**Comment by themrmax** [comment]
@jreback lol i don't know, maybe some people use series without ever indexing them? otherwise do you think we should add a check for hashable values?

**Comment by jreback** [comment]
```
In [4]: s = pd.Series ([1,2,3], index = ([1],[2],[3]))

In [5]: s.loc[[1]]
TypeError: unhashable type: 'list'

In [7]: s.index.inferred_type
Out[7]: 'mixed'
```

So we do actually check hashable, but its fairly lazy as its somewhat to expensive to do this. The point is that this is non-idiomatic, and though may work in parts of pandas is not supported, nor recommended in any way.

So if you'd like to update with a stronger warning would be great.

**Comment by jorisvandenbossche** [comment]
@themrmax Can you update this? (leave out the hashable change for now, but you can keep the change regarding uniqueness)

File: pandas/core/series.py [hunk 1]
```diff
@@ -117,11 +117,9 @@ class Series(base.IndexOpsMixin, strings.StringAccessorMixin,
     data : array-like, dict, or scalar value
         Contains data stored in Series
     index : array-like or Index (1d)
-        Values must be unique and hashable, same length as data. Index
-        object (or other iterable of same length as data) Will default to
-        RangeIndex(len(data)) if not provided. If both a dict and index
-        sequence are used, the index will override the keys found in the
-        dict.
+	Values must be hashable and the same length as data. Will default to
```
Comment (jreback) [inline pandas/core/series.py]: spacing here

Comment (themrmax) [inline pandas/core/series.py]: ah sorry about just learned how to use `textwidth` in vim

File: pandas/core/series.py [hunk 2]
```diff
@@ -117,11 +117,9 @@ class Series(base.IndexOpsMixin, strings.StringAccessorMixin,
     data : array-like, dict, or scalar value
         Contains data stored in Series
     index : array-like or Index (1d)
-        Values must be unique and hashable, same length as data. Index
-        object (or other iterable of same length as data) Will default to
-        RangeIndex(len(data)) if not provided. If both a dict and index
-        sequence are used, the index will override the keys found in the
-        dict.
+        Values must be hashable and the same length as data. Will default to
+        RangeIndex(len(data)) if not provided. If both a dict and index sequence
```
Comment (jorisvandenbossche) [inline pandas/core/series.py]: This line is too long (therefore travis is failing, we test for PEP8 (line length < 80))

**Comment by jorisvandenbossche** [comment]
@themrmax Thanks! 
(I added one sentence from another PR for the same issue ("Non-unique index values are allowed") before merging)

**Comment by themrmax** [comment]
cool thanks @jorisvandenbossche yes that's clearer now !

**Pull request merged**